addons/binding/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/internal/ChromecastDiscoveryParticipant.java

```diff
@@ -11,79 +11,78 @@
 import java.util.Map;
 import java.util.Set;
 
+import javax.jmdns.ServiceInfo;
+
 import org.eclipse.smarthome.config.discovery.DiscoveryResult;
 import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
-import org.eclipse.smarthome.config.discovery.UpnpDiscoveryParticipant;
 import org.eclipse.smarthome.core.thing.ThingTypeUID;
 import org.eclipse.smarthome.core.thing.ThingUID;
-import org.jupnp.model.meta.DeviceDetails;
-import org.jupnp.model.meta.ManufacturerDetails;
-import org.jupnp.model.meta.ModelDetails;
-import org.jupnp.model.meta.RemoteDevice;
-import org.jupnp.model.meta.RemoteDeviceIdentity;
+import org.eclipse.smarthome.io.transport.mdns.discovery.MDNSDiscoveryParticipant;
 import org.openhab.binding.chromecast.ChromecastBindingConstants;
 
 /**
  * The {@link ChromecastDiscoveryParticipant} is responsible for discovering Chromecast devices through UPnP.
  *
  * @author Kai Kreuzer - Initial contribution
+ * @author Daniel Walters - Change discovery protocol to mDNS
  *
  */
-public class ChromecastDiscoveryParticipant implements UpnpDiscoveryParticipant {
+public class ChromecastDiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    private static final String SERVICE_TYPE = "_googlecast._tcp.local.";
 
     @Override
     public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
         return ChromecastBindingConstants.SUPPORTED_THING_TYPES_UIDS;
     }
 
     @Override
-    public DiscoveryResult createResult(final RemoteDevice device) {
-        final ThingUID uid = getThingUID(device);
+    public String getServiceType() {
+        return SERVICE_TYPE;
+    }
+
+    @Override
+    public DiscoveryResult createResult(ServiceInfo service) {
+
+        final ThingUID uid = getThingUID(service);
         if (uid == null) {
             return null;
         }
 
         final Map<String, Object> properties = new HashMap<>(2);
-        properties.put(ChromecastBindingConstants.HOST, device.getDetails().getBaseURL().getHost());
-        properties.put(ChromecastBindingConstants.SERIAL_NUMBER, device.getDetails().getSerialNumber());
+        String host = service.getHostAddresses()[0];
+        properties.put(ChromecastBindingConstants.HOST, host);
+
+        String friendlyName = service.getPropertyString("fn"); // friendly name;
 
-        final DiscoveryResult result = DiscoveryResultBuilder.create(uid).withThingType(getThingType(device))
-                .withProperties(properties).withLabel(device.getDetails().getFriendlyName())
+        final DiscoveryResult result = DiscoveryResultBuilder.create(uid).withThingType(getThingType(service))
+                .withProperties(properties).withLabel(friendlyName)
                 .withRepresentationProperty(ChromecastBindingConstants.SERIAL_NUMBER).build();
```

Comment:
No, because you are specifying a representation property, which you do not set. If you can get this information, set the property. Otherwise, remove the call to withRepresentationProperty (or find another property, which might be usable).